command/server.go

```diff
@@ -1258,57 +1258,29 @@ func (c *ServerCommand) Run(args []string) int {
 	}
 
 	ctx := context.Background()
-	existingSealGenerationInfo, err := vault.PhysicalSealGenInfo(ctx, backend)
-	if err != nil {
-		c.UI.Error(fmt.Sprintf("Error getting seal generation info: %v", err))
-		return 1
-	}
 
-	hasPartialPaths, err := hasPartiallyWrappedPaths(ctx, backend)
-	if err != nil {
-		c.UI.Error(fmt.Sprintf("Cannot determine if there are partially seal wrapped entries in storage: %v", err))
-		return 1
-	}
-	setSealResponse, err := setSeal(c, config, infoKeys, info, existingSealGenerationInfo, hasPartialPaths)
+	setSealResponse, secureRandomReader, err := c.configureSeals(ctx, config, backend, infoKeys, info)
 	if err != nil {
 		c.UI.Error(err.Error())
 		return 1
 	}
-	if setSealResponse.sealConfigWarning != nil {
-		c.UI.Warn(fmt.Sprintf("Warnings during seal configuration: %v", setSealResponse.sealConfigWarning))
-	}
 
-	for _, seal := range setSealResponse.getCreatedSeals() {
-		seal := seal // capture range variable
-		// Ensure that the seal finalizer is called, even if using verify-only
+	if setSealResponse.barrierSeal != nil {
```

Comment:
The code to call `Finalize` no longer uses `SetSealResponse.getCreatedSeals()` here, but the operator diagnose still does. Moreover, there is now a third place where the finalize code is called (in `Command.Run`). 

What about replacing `SetSealResponse.getCreatedSeals()` by a method such as `SetSealResponse.finalizeSeals()`, and calling defer on that instead?

Reply:
I don't think I actually meant to get rid of the loop here, I just forgot to change it back when I was fixing the return values for `configureSeals`. I tried out adding the `finalizeSeals` function, but I think it makes error handling harder